Source/Particles/ParticleCreation/SmartCreate.H

```diff
@@ -47,32 +47,32 @@ struct SmartCreate
         const int id = 0) const noexcept
     {
 #if defined(WARPX_DIM_3D)
-        prt.m_aos[i_prt].pos(0) = x;
-        prt.m_aos[i_prt].pos(1) = y;
-        prt.m_aos[i_prt].pos(2) = z;
+        prt.m_rdata[0][i_prt] = x;
```

Comment:
```suggestion
        prt.m_rdata[PIdx::x][i_prt] = x;
```

This was used in BTDParticleFunctor, so it can be used here to